numerical_methods/roots.c

```diff
@@ -0,0 +1,166 @@
+#include <stdio.h>
+int value(double arr[], double m, double *e, int n)
+{
+    int b = n + 1;
+    int a = 0;
+    int c = 0;
+    double sum = 0;
+    double d = 1;
+    int i = n;
+    while (i >= 0)
+    {
+        while (c < a)
+        {
+            d = d * m;
+            c++;
+        }
+        sum = arr[i] * d + sum;
+        a++;
+        i--;
+    }
+    *e = sum;
+}
+int diff(double arr[], double m, double *c, int n)
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.